python/pipeline/utils/DLC_tools.py

```diff
@@ -0,0 +1,1033 @@
+
+        fitted_core_dict = self.fitted_core(frame_num)
+
+        if fitting_method == 'circle':
+
+            circle_frame = ax.imshow(fitted_core_dict['circle_fit']['frame'])
+            circle_mask = ax.imshow(
+                fitted_core_dict['circle_visible']['mask'], alpha=0.2)
+
+        elif fitting_method == 'ellipse':
+
+            ellipse_frame = ax.imshow(fitted_core_dict['ellipse_fit']['frame'])
+            ellipse_mask = ax.imshow(
+                fitted_core_dict['ellipse_visible']['mask'], alpha=0.2)
+
+        else:
+            raise ValueError(
+                'fitting method must be either an ellipse or a circle!')
+
+        plt.title('frame num: ' + str(frame_num), fontsize=30)
+
+        plt.axis('off')
+        plt.tight_layout()
+
+        if save_fig:
+            plt.savefig(os.path.join(
+                self.compressed_cropped_dir_path, 'fitted_frame_' + str(frame_num) + '.png'))
+
+    def plot_fitted_multi_frames(self, start, end, fitting_method='circle', save_gif=False):
+
+        fig, ax = self.configure_plot()
+
+        plt_list = []
+
+        for frame_num in range(start, end):
+
+            self.plot_fitted_frame(frame_num=frame_num, ax=ax, fitting_method=fitting_method)
+
+            fig.canvas.draw()
+
+            data = np.fromstring(fig.canvas.tostring_rgb(),
+                                 dtype=np.uint8, sep='')
+            data = data.reshape(fig.canvas.get_width_height()[::-1] + (3,))
+            plt_list.append(data)
+
+            display.clear_output(wait=True)
+            display.display(pl.gcf())
+            time.sleep(0.5)
+
+            plt.cla()
+
+        if save_gif:
+            gif_path = self.video_path.split('.')[0] + '_fitted_' + \
+                str(start) + '_' + str(end) + '.gif'
+            imageio.mimsave(gif_path, plt_list, fps=1)
+
+        plt.close('all')
+
+    def make_movie(self, start, end, fitting_method='circle'):
+
+        import matplotlib.animation as animation
+
+        # initlize with start frame
+        fig, ax = self.configure_plot()
+
+        self.plot_fitted_frame(frame_num=start, ax=ax, fitting_method=fitting_method)
+
+        def update_frame(frame_num):
+
+            # clear out the axis
+            plt.cla()
+            
+            self.plot_fitted_frame(frame_num=frame_num, ax=ax, fitting_method=fitting_method)
+
+        ani = animation.FuncAnimation(fig, update_frame, range(
+            start+1, end))  # , interval=int(1/self.clip.FPS)
+        # ani = animation.FuncAnimation(fig, self.plot_fitted_frame, 10)
+        writer = animation.writers['ffmpeg'](fps=self.clip.FPS)
+
+        # dpi=self.dpi, fps=self.clip.FPS
+        save_video_path = self.video_path.split('.')[0] + '_fitted_' + \
+                str(start) + '_' + str(end) + '_labeled.avi'
+        ani.save(save_video_path, writer=writer, dpi=self.dpi)
+
+        return ani
+
+
+# # We just subclass Rectangle so that it can be called with an Axes
```

Comment:
do you need all of these comments?, they are in your git history so they are there if you wanna come back to them but may not need them in the main repo

Reply:
I deleted all these per request